projects/storefrontlib/src/lib/ui/layout/main/storefront.component.ts

```diff
@@ -1,12 +1,15 @@
 import { Component, OnInit } from '@angular/core';
+import { ConfigurableRoutesService } from '@spartacus/core';
 
 @Component({
   selector: 'cx-storefront',
   templateUrl: './storefront.component.html',
   styleUrls: ['./storefront.component.scss']
 })
 export class StorefrontComponent implements OnInit {
-  constructor() {}
+  constructor(private configurableRoutesService: ConfigurableRoutesService) {}
```

Comment:
Can we replace this with the APP_INITIALIZER?

Reply:
I use APP_INITIALIZER here: https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/215/files/afc5694ccf1f37356ddd2391b832358ceb21ee0d#diff-d24fccac232afa79f50bbe09530ef628
1. I provide APP_INITIALIZER using factory that returns a promise. The promise resolves eventually when translations config is fetched (or right immediate, if there is no url configured for translations config)
2. This factory depends on ConfigurableRoutesLoader class which performs this fetch and merges the result with default and statically configured translations config... and stores the result itself.
3. ConfigurableRoutesService depends on ConfigurableRoutesLoader - it gets the final merged translations config and uses for performing translations of real routes
4. StorefrontComponent depends on ConfigurableRoutesService - it will just pass active language to service on every language change. (and by depending on the service, it forces it to initialize)

So StorefrontComponent depends transitively on APP_INITIALIZER.